doc/docs/2.3.x/how-tos/basic-data-operations/sql-ingest.md

```diff
@@ -1,251 +1,232 @@
 
-!!! Note
-    `pachctl update pipeline` will create pipelines if none exist, or update your existing pipelines otherwise.
+- **1 Input Data Repo**: Used to store timestamp files at the cronSpec's set interval rate (`--arg cronSpec="pullInterval" \`) to trigger the pipeline.
+- [**1 Cron Pipeline**](../../../concepts/pipeline-concepts/pipeline/cron/#cron-pipeline): Houses the spec details that define the input type and settings and  data transformation.
+-  **1 Output Repo**: Used to store the data transformed by the cron pipeline; set by the pipeline spec's `pipeline.name` attribute, which you can define through the Jsonnet parameter `--arg name=outputRepoName \`.
+- **1 Output File**: Used to save the query results (JSON or CSV) and potentially be used as input for a following pipeline.
 
+![sql-ingest-diagram](../images/sql-ingest-cron-example.png)
 
-When the command is run, the database will be queried on a schedule defined in your `cronSpec` parameter and a result file committed to the output repo named after `name`.
+In the default Jsonnet template, the file generated is obtainable from the output repo, `outputRepoName@master:/0000`. The filename is hardcoded, however you could paramaterize this as well using a custom Jsonnet pipeline spec and passing `--arg outputFile='0000'`. The file's contents are the result of the query(`--arg query="query"`) being ran against the database`--arg url="connectionStringToDdatabase"` ; both are defined in the `transform.cmd` attribute.
 
-### Database Secret
-Before you create your SQL Ingest pipelines, make sure to create a [generic secret](../../advanced-data-operations/secrets/#create-a-secret) containing your database password in the field `PACHYDERM_SQL_PASSWORD`.
 
-!!! Example
-    ```yaml
-    apiVersion: v1
-    kind: Secret
-    metadata:
-      name: mysql-creds
-    data:
-      "PACHYDERM_SQL_PASSWORD": "cm9vdA==" # base64 encoded
-    ```
+### About SQL Ingest Pipeline Specs
 
-!!! Info "TL;DR"
-    - Run the following command to generate your secret:
+To create an SQL Ingest Jsonnet Pipeline spec, you must have a `.jsonnet` file and several parameters:
 
-        `kubectl create secret generic <secret-name>  --from-literal=PACHYDERM_SQL_PASSWORD=<password-to-warehouse> --dry-run=client --output=json > yourwarehousesecret.json`
+```shell
+pachctl update pipeline --jsonnet <https://your-SQL-ingest-pipeline-spec.jsonnet> \
+  --arg name=<outputRepoName> \
+  --arg url="<connectionStringToDdatabase>" \
+  --arg query="<query>" \
+  --arg hasHeader=<boolean> \
+  --arg cronSpec="<pullInterval>" \
+  --arg secretName="<secretName>" \
+  --arg format=<CsvOrJson> 
+```
 
-    - Then apply it to your Pachyderm cluster:
+- The name of each pipeline (and their related input/output repos) are derived from the `name` parameter (`--arg name=outputRepoName`).
+- The same base image [pachctf](https://hub.docker.com/repository/docker/pachyderm/pachtf){target=_blank} is used in both pipelines.
```

Comment:
This is no longer true. We are getting rid of the pachtf *docker image* in favour of injecting the pachtf binary into any user container at pod inititalization.